test/decorator-annotator_test.ts

```diff
@@ -28,68 +30,82 @@ function sources(sourceText: string): Map<string, string> {
-      it('generates a source map', () => {
-        const {output, sourceMap} = translate(`
-/** @Annotation */ let Test1: Function;
-@Test1
-export class Foo {
-}
-let X = 'a string';`);
-        const rawMap = sourceMap.toJSON();
-        const consumer = new SourceMapConsumer(rawMap);
-        const lines = output.split('\n');
-        const stringXLine = lines.findIndex(l => l.indexOf('a string') !== -1) + 1;
-        expect(consumer.originalPositionFor({line: stringXLine, column: 10}).line)
-            .to.equal(6, 'string X definition');
-      });
-
-      describe('class decorator rewriter', () => {
-        it('leaves plain classes alone', () => {
-          expectUnchanged(`class Foo {}`);
-        });
-
-        it('leaves un-marked decorators alone', () => {
-          expectUnchanged(`
-          let Decor: Function;
+    const transformerHost: tsickle.TsickleHost = {
+      shouldSkipTsickleProcessing: (filePath) => !sources(sourceText).has(filePath),
+      pathToModuleName: cliSupport.pathToModuleName,
+      shouldIgnoreWarningsForPath: (filePath) => false,
+      fileNameToModuleId: (filePath) => filePath,
+      transformDecorators: true,
+      googmodule: true,
+      es5Mode: false,
+      untyped: false,
+    };
+
+    const files = new Map<string, string>();
+    const {diagnostics} = tsickle.emitWithTsickle(
+        program, transformerHost, host, compilerOptions, undefined, (path, contents) => {},
+        undefined, undefined, {beforeTs: [generateAstPrintingTransform(files)]});
+
+    if (!allowErrors) {
+      // tslint:disable-next-line:no-unused-expression
+      expect(diagnostics, tsickle.formatDiagnostics(diagnostics)).to.be.empty;
+    }
+
+    return {output: files.get(testCaseFileName)!, diagnostics};
+  }
+
+  function prettyPrint(sourceText: string) {
+    return ts.createPrinter().printFile(
+        ts.createSourceFile('', sourceText, compilerOptions.target!));
+  }
+
+  function normalizeQuotes(sourceText: string) {
+    return sourceText.replace(/"/g, `'`);
+  }
+
+  function expectUnchanged(sourceText: string) {
+    expectEqual(sourceText, sourceText);
+  }
+
+  function expectEqual(sourceText: string, expected: string) {
+    expect(normalizeQuotes(translate(sourceText).output))
+        .to.equal(normalizeQuotes(prettyPrint(expected)));
+  }
+
+  describe('class decorator rewriter', () => {
+    it('leaves plain classes alone', () => {
+      expectUnchanged(`class Foo {}`);
+    });
+
+    it('leaves un-marked decorators alone', () => {
+      expectUnchanged(`
+          let Decor: any;
           @Decor class Foo {
             constructor(@Decor p: number) {}
             @Decor m(): void {}
           }`);
-        });
+    });
 
-        it('transforms decorated classes', () => {
-          expect(translate(`
-/** @Annotation */ let Test1: Function;
-/** @Annotation */ let Test2: Function;
+    it('transforms decorated classes', () => {
+      expectEqual(`
+/** @Annotation */ let Test1: any;
```

Comment:
it might be worth documenting somewhere here that we use `any` to fake out a decorator. Maybe you could define `type FakeDecorator = any` in the .d.ts file in sources above, and leave a comment there?

Reply:
Done.